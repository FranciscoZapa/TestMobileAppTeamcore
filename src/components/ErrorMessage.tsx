import React from 'react';
import {Box, HStack, Heading} from 'native-base';
import {errorlabel} from '../utils/constants';
import {styles} from '../styles/status/ErrorMessageStyles';

export const ErrorMessage = (): JSX.Element => (
  <Box
    alignItems="center"
    justifyContent="space-evenly"
    height={'100%'}
    style={styles.box}>
    <HStack space={2} justifyContent="center">
      <Heading color="white" fontSize="md">
        {errorlabel}
      </Heading>
    </HStack>
  </Box>
);
