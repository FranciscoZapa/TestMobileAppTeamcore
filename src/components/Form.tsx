import React, {PropsWithChildren} from 'react';
import {FlatList, Text} from 'react-native';
import {Question} from './Question';

type FormProps = PropsWithChildren<{
  data: Array<{
    answers: Array<{answer: string; answer_id: string}> | null;
    question: string;
    question_id: string;
  }> | null;
}>;

export const Form = ({data}: FormProps): JSX.Element => {
  if (data && data.length === 0) {
    return <Text>no hay</Text>;
  }
  return (
    <FlatList
      data={data}
      renderItem={({item}) => (
        <Question
          question={item.question}
          answers={item.answers}
          id={item.question_id}
        />
      )}
      keyExtractor={item => item.question_id}
    />
  );
};
