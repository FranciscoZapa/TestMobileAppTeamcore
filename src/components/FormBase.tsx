import React, {PropsWithChildren} from 'react';
import {Box} from 'native-base';

type FormBaseProps = PropsWithChildren<{}>;

export const FormBase = ({children}: FormBaseProps): JSX.Element => {
  return (
    <Box alignItems="center" justifyContent="flex-start" height={'75%'}>
      {children}
    </Box>
  );
};
