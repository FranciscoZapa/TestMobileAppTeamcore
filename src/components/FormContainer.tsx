import React, {PropsWithChildren} from 'react';
import {Box} from 'native-base';
import {styles} from '../styles/form/FormStyles';

type FormContainerProps = PropsWithChildren<{}>;

export const FormContainer = ({children}: FormContainerProps): JSX.Element => {
  return (
    <Box
      maxH={'99%'}
      maxW={'100%'}
      rounded="lg"
      overflow="hidden"
      borderColor="brand.700"
      padding={'3'}
      style={styles.form}>
      {children}
    </Box>
  );
};
