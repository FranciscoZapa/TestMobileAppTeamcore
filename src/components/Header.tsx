import React from 'react';
import {Box, Heading, Text} from 'native-base';
import {HeaderBase} from './HeaderBase';
import {styles} from '../styles/header/HeaderBase';
import {description, welcome} from '../utils/constants';

export const Header = (): JSX.Element => {
  return (
    <HeaderBase>
      <Heading style={styles.title}>
        <Text style={styles.text}>{welcome}</Text>
      </Heading>
      <Box style={styles.description}>
        <Text style={styles.text}>{description}</Text>
      </Box>
    </HeaderBase>
  );
};
