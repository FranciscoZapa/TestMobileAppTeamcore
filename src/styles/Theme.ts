// 1. Import the extendTheme function
import {extendTheme} from 'native-base';
// 2. Extend the theme to include custom colors, fonts, etc
const newColorTheme = {
  brand: {
    900: '#002b5a',
    800: '#003165',
    700: '#003670',
    600: '#1a4a7e',
    500: '#335e8d',
  },
  text: {
    white: '#ffffff',
  },
  background: {
    varyDark: '#2E4453',
    dark: '#002635',
    light: '#e6ebf1',
  },
  general: {
    lightBlue: '#005B9A',
    grayBlue: '#B8D9E9',
  },
};
export const theme = extendTheme({colors: newColorTheme});

export const mainColor: string = '#003670';
