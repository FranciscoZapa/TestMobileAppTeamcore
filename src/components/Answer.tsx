import React, {PropsWithChildren} from 'react';
import {View, Text} from 'react-native';
import {styles} from '../styles/FormScreenStyles';

type QuestionProps = PropsWithChildren<{
  answer: string;
  id: string;
}>;

export const Answer = ({answer, id}: QuestionProps): JSX.Element => {
  return (
    <View style={styles.sectionContainer} id={id}>
      <Text style={[styles.sectionTitle]}>{answer}</Text>
    </View>
  );
};
