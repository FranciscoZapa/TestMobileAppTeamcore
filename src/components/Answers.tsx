import React, {PropsWithChildren} from 'react';
import {FlatList} from 'react-native';
import {AnswerStruct} from '../types/form/formTypes';
import {Answer} from './Answer';

type AnswersProps = PropsWithChildren<{
  answers: Array<AnswerStruct> | null;
  id: string;
}>;

export const Answers = ({answers, id}: AnswersProps): JSX.Element => {
  return (
    <FlatList
      data={answers}
      renderItem={({item}) => (
        <Answer
          answer={item.answer}
          answer_id={item.answer_id}
          question_id={id}
        />
      )}
      keyExtractor={item => item.answer_id}
    />
  );
};
