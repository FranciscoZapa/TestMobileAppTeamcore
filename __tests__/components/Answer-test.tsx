import React from 'react';
import {useSelector} from 'react-redux';
import {NativeBaseProvider} from 'native-base';
import renderer from 'react-test-renderer';
import {theme} from '../../src/styles/Theme';
import {Answer} from '../../src/components/Answer';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe('Answer', () => {
  it('should render the text of the answer', () => {
    (useSelector as jest.Mock).mockReturnValue({
      form: {
        answers: [{question_id: '1', answer_id: '1'}],
      },
      answers: [{question_id: '1', answer_id: '1'}],
    });
    const tree = renderer
      .create(
        <NativeBaseProvider theme={theme}>
          <Answer answer="Answer 1" answer_id="1" question_id="1" />
        </NativeBaseProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
