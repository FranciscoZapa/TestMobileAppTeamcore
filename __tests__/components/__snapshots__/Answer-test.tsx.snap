// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer should render the text of the answer 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
